packages/m2m-gateway/src/services/tenantService.ts

```diff
@@ -1,8 +1,46 @@
+import { WithLogger } from "pagopa-interop-commons";
+import { m2mGatewayApi } from "pagopa-interop-api-clients";
 import { PagoPAInteropBeClients } from "../clients/clientsProvider.js";
+import { M2MGatewayAppContext } from "../utils/context.js";
+import { toM2MTenant } from "../api/tenantApiConverter.js";
 
 export type TenantService = ReturnType<typeof tenantServiceBuilder>;
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function tenantServiceBuilder(_clients: PagoPAInteropBeClients) {
-  return {};
+export function tenantServiceBuilder({
+  tenantProcessClient,
+}: PagoPAInteropBeClients) {
+  return {
+    getTenants: async (
+      { logger, headers }: WithLogger<M2MGatewayAppContext>,
+      queryParams: m2mGatewayApi.GetTenantsQueryParams
+    ): Promise<m2mGatewayApi.Tenants> => {
+      const { externalIdOrigin, externalIdValue, limit, offset } = queryParams;
+
+      logger.info(
+        `Retrieving tenants for externalIdOrigin ${externalIdOrigin} externalIdValue ${externalIdValue} limit ${limit} offset ${offset}`
+      );
+
+      const {
+        data: { results, totalCount },
+      } = await tenantProcessClient.tenant.getTenants({
+        queries: {
+          externalIdOrigin,
+          externalIdValue,
+          limit,
+          offset,
+        },
```

Comment:
Same comment I left in other PR https://github.com/pagopa/interop-be-monorepo/pull/1814/files#r2075625608